games/devtest/mods/unittests/metadata.lua

```diff
@@ -77,6 +77,29 @@ local function test_metadata(meta)
 	assert(not meta:equals(compare_meta))
 end
 
+local function test_metadata_compat(meta)
+	-- key/value removal using set_string (undocumented, deprecated way)
+	meta:set_string("key", "value")
+	assert(meta:get_string("key") == "value")
+	meta:set_string("key", nil) -- ignore warning
+	assert(meta:to_table().fields["key"] == nil)
```

Comment:
cleaner imo
```suggestion
	assert(not meta:contains("key"))
```

Reply:
This is to check the `to_table()` functionality. Setters + `to_table` were already tested previously but not removing keys. It *should not* make a difference after all, but I found this to be a good addition to the already existing checks.